### PR TITLE
[FLINK-29319][table] Upgrade Calcite to 1.32.0

### PR DESCRIPTION
**Pull request by snuyanzin**
## What is the purpose of the change

This is upgrade of Calcite to 1.32.0
since there is a small changeset it is in one commit
there are 2 Calcite's classes to be removed

also in Calcite 1.32.0 ESRI spatial engine was replaced JTS and proj4j (https://issues.apache.org/jira/browse/CALCITE-4294)

## Brief change log

 - Removed 
 _flink-table/flink-table-planner/src/main/java/org/apache/calcite/rel/core/Snapshot.java_ ,
_flink-table/flink-table-planner/src/main/java/org/apache/calcite/rel/hint/NodeTypeHintPredicate.java_
- Synced with Calcite [release-1.32.0 tag](https://github.com/apache/calcite/releases/tag/calcite-1.32.0)
  _flink-table/flink-table-planner/src/main/java/org/apache/calcite/sql/validate/SqlValidatorImpl.java_,
  _flink-table/flink-table-planner/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java_
- NOTICE, pom files

## Verifying this change

This change is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: ( no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): ( no )
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: ( no )
  - The S3 file system connector: ( no)

## Documentation

  - Does this pull request introduce a new feature? ( no)
  - If yes, how is the feature documented? (not applicable )

### CODE REVIEW

File: flink-table/flink-table-planner/pom.xml [hunk 1]
```diff
@@ -300,7 +300,9 @@ under the License.
 							<include>org.apache.calcite.avatica:*</include>
 
 							<!-- Calcite's dependencies -->
-							<include>com.esri.geometry:esri-geometry-api</include>
+							<include>org.locationtech.proj4j:proj4j</include>
```
Comment (twalthr) [inline flink-table/flink-table-planner/pom.xml:303]: Can we exclude those dependencies? Which classes actually need them?

**Review by twalthr — CHANGES_REQUESTED** [review]
Let's try to keep the deps minimal. After that we should be good to merge this.

File: flink-table/flink-table-planner/pom.xml (hunk 1 above)
Comment (snuyanzin) [reply flink-table/flink-table-planner/pom.xml]: Thanks for the pointing to that.
Yep, it seems `org.locationtech.proj4j:proj4j` and `org.locationtech.jts.io:jts-io-common` could be omitted.
`org.locationtech.jts:jts-core`  is used in `RexBuilder`, `RexLiteral` and other classes like 
https://github.com/apache/calcite/blob/413eded693a9087402cc1a6eefeca7a29445d337/core/src/main/java/org/apache/calcite/rex/RexBuilder.java#L62
so it's required

**Review by twalthr — APPROVED** [review]
LGTM thanks @snuyanzin.

**Pull request merged**